tests/integration_tests/conftest.py

```diff
@@ -119,49 +129,56 @@ def get_validated_source(
 
 
 @pytest.fixture(scope="session")
-def setup_image(session_cloud: IntegrationCloud, request):
-    """Setup the target environment with the correct version of cloud-init.
+def setup_image(session_cloud: IntegrationCloud):
+    """Set up the target environment with the correct version of cloud-init.
 
     So we can launch instances / run tests with the correct image
     """
-    source = get_validated_source(session_cloud)
-    if not (
-        source.installs_new_version()
-        or integration_settings.INCLUDE_COVERAGE
-        or integration_settings.INCLUDE_PROFILE
-    ):
-        return
-    log.info("Setting up source image")
-    client = session_cloud.launch()
-    if source.installs_new_version():
-        log.info("Installing cloud-init from %s", source.name)
-        client.install_new_cloud_init(source)
-    if (
-        integration_settings.INCLUDE_PROFILE
-        and integration_settings.INCLUDE_COVERAGE
-    ):
-        log.error(
-            "Invalid configuration, cannot enable both profile and coverage."
+    try:
+        source = get_validated_source(session_cloud)
+        if not (
+            source.installs_new_version()
+            or integration_settings.INCLUDE_COVERAGE
+            or integration_settings.INCLUDE_PROFILE
+        ):
+            yield
+            return
+        log.info("Setting up source image")
+        client = session_cloud.launch()
+        if source.installs_new_version():
+            log.info("Installing cloud-init from %s", source.name)
+            client.install_new_cloud_init(source)
+        if (
+            integration_settings.INCLUDE_PROFILE
+            and integration_settings.INCLUDE_COVERAGE
+        ):
+            log.error(
+                "Invalid configuration, cannot enable both profile and "
+                "coverage."
+            )
+            raise ValueError()
+        if integration_settings.INCLUDE_COVERAGE:
+            log.info("Installing coverage")
+            client.install_coverage()
+        elif integration_settings.INCLUDE_PROFILE:
+            log.info("Installing profiler")
+            client.install_profile()
+        # All done customizing the image, so snapshot it and make it global
+        snapshot_id = client.snapshot()
+        client.cloud.snapshot_id = snapshot_id
+        # Even if we're keeping instances, we don't want to keep this
+        # one around as it was just for image creation
+        client.destroy()
+        log.info("Done with environment setup")
+
+        # For some reason a yield here raises a
+        # ValueError: setup_image did not yield a value
+        # during setup so use a finalizer instead.
```

Comment:
we lost the `request.addfinalizer` line that used to be under this comment (and can likely now be replaced with a yield followed by `delete_snapshot`).